sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -72,4 +75,55 @@ export class EducationProgramOfferingController {
       offeringDelivered: offering.offeringDelivered,
     }));
   }
+
+  @HasLocationAccess("locationId")
+  @Get("location/:locationId/education-program/:programId/offering/:offeringId")
+  async getProgramOffering(
+    @Param("locationId") locationId: number,
+    @Param("programId") programId: number,
+    @Param("offeringId") offeringId: number,
+  ): Promise<ProgramOfferingDto> {
+    //To retrive Education program offering corresponding to ProgramId and LocationId
+    const offering = await this.programOfferingService.getProgramOffering(
+      locationId,
+      programId,
+      offeringId,
+    );
+    if (!offering) {
+      throw new UnprocessableEntityException(
+        "Not able to find a Education Program Offering associated with the current Education Program, Location and offering.",
+      );
+    }
+    return offering;
```

Comment:
I believe that the entire object from DB entity will be returned here. The DTO will not prevent that the fields that are not part of the DTO be returned.

Reply:
Please elaborate on this, I got only the DTO values that are returned.

Reply:
`offering` is a direct DB entity, while returning we may need to prepare the DTO and return

Reply:
done